.build/tasks/Common.Functions.psm1

```diff
@@ -105,7 +111,20 @@ function Get-ModuleVersion
             a version string with metadata in the CI pipeline that can look like
             this: 1.15.0-pr0224-0022+Sha.47ae45eb2cfed02b249f239a7c55e5c71b26ab76.Date.2020-01-07
         #>
-        $moduleVersion = ($moduleVersion -split '\+', 2)[0]
+        $moduleVersionWithoutMetadata = ($ModuleVersion -split '\+', 2)[0]
+
+        <#
+            Parses the prerelease string to remove any suffix that is not
+            handled by the cmdlet Publish-Module.
+        #>
+        $moduleVersion, $preReleaseString = $moduleVersionWithoutMetadata -split '-', 2
+
+        if ($preReleaseString)
```

Comment:
non-blocking: I would have added a comment saying the error is from Publish-Module which does not support semver compliant pre-release strings.